FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs

```diff
@@ -366,8 +366,14 @@ private async Task RenewJobLockLeaseAsync(CancellationToken cancellationToken =
             }
             catch
             {
-                _logger.LogError("Failed to renew job lease.");
-                throw;
+                if (_renewLockTimer.Enabled)
+                {
+                    _logger.LogError("Failed to renew job lease.");
+                    throw;
+                }
+
+                _logger.LogInformation("Refresh timer already stopped.");
```

Comment:
When will this case happen? In future distributed system?

Reply:
Corner cases found in E2E tests